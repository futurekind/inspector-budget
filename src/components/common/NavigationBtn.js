import React, { PropTypes } from 'react';
import styled from 'styled-components';

import { colors, sizes } from '../../utils/styles';

const Button = styled.button`
    width: 100%;
    height: ${sizes.navigationWidth}px;
    border: none;
    border-top: 1px solid ${colors.dark};
    border-bottom: 1px solid ${colors.light};
    display: block;
    position: relative;
    color: #fff;
    background: none;

    span {
        width: 100%;
        position: absolute;
        left: 0;
        top: 50%;
        transform: translateY(-50%);
    }
`

const NavigationBtn = ({
    children,
    onClick
}) => {
    return (
        <Button onClick={ onClick }>
            <span>{ children }</span>
        </Button>
    )
}

NavigationBtn.propTypes = {
    children: PropTypes.any,
    icon: PropTypes.string,
}

export default NavigationBtn