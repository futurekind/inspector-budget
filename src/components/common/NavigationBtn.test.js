import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NavigationBtn from './NavigationBtn';

describe('NavigationBtn', () => {

    it('renders correctly', () => {
        const wrapper = shallow(<NavigationBtn to="" />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders icon when passed as prop', () => {
        const wrapper = shallow(<NavigationBtn to="" icon="my-icon" />)
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders as active when passed as prop', () => {
        const wrapper = shallow(<NavigationBtn to="" active />)
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('renders its children', () => {
        const wrapper = shallow(<NavigationBtn to="">Label</NavigationBtn>)
        expect(toJSON(wrapper)).toMatchSnapshot();
    })

    it('delegates onClick prop', () => {
        const spy = jest.fn();
        const wrapper = shallow(<NavigationBtn to="" onClick={spy} />)
        const button = wrapper.find('button');
        wrapper.simulate('click');

        expect(spy).toHaveBeenCalled()
    })

})