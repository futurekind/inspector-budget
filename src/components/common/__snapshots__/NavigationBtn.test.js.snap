exports[`NavigationBtn renders as active when passed as prop 1`] = `
<styled.button
  active={true}>
  <styled.span />
</styled.button>
`;

exports[`NavigationBtn renders correctly 1`] = `
<styled.button>
  <styled.span />
</styled.button>
`;

exports[`NavigationBtn renders icon when passed as prop 1`] = `
<styled.button>
  <styled.span>
    <styled.i
      className="material-icons">
      my-icon
    </styled.i>
  </styled.span>
</styled.button>
`;

exports[`NavigationBtn renders its children 1`] = `
<styled.button>
  <styled.span>
    Label
  </styled.span>
</styled.button>
`;
